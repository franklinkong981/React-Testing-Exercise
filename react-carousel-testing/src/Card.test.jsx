import { render, fireEvent } from "@testing-library/react";
import Card from "./Card.jsx";
import TEST_IMAGES from "./_testCommon.js";

//Smoke test
it("renders the Card component without crashing", () => {
  render(<Card caption={TEST_IMAGES[0].caption} src={TEST_IMAGES[0].src} currNum={1} totalNum={TEST_IMAGES.length}/>);
});

//snapshot test
it("matches the snapshot of the Card component", function() {
	const {asFragment} = render(<Card caption={TEST_IMAGES[0].caption} src={TEST_IMAGES[0].src} currNum={1} totalNum={TEST_IMAGES.length}/>);
	expect(asFragment()).toMatchSnapshot();
});