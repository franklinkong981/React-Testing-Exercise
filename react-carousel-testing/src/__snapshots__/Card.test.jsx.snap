// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot of the Card component 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;
